sycl/include/sycl/ext/oneapi/property_list/property_value.hpp

```diff
@@ -0,0 +1,105 @@
+//==------ property_value.hpp --- SYCL compile-time property values --------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// HOW-TO: Add new compile-time property
+//  1. Define property class with `value_t` that must be `property_value` with
+//     the first template argument being the property class itself.
+//  2. Add a new enumerator to `sycl::ext::oneapi::detail::CompileTimePropKind`
+//     representing the new property. Increment
+//     `sycl::ext::oneapi::detail::CompileTimePropKind::CompileTimePropKindSize`
+//  3. Specialize `sycl::ext::oneapi::detail::CompileTimePropertyToKind` for the
+//     new property class. The specialization should have a `value` member
+//     with the value equal to the enumerator added in 2.
+//  4. Add an `inline constexpr` variable in the same namespace as the property.
+//     The variable should have the same type as `value_t` of the property class
+//     and should be named as the property class with `_v` appended, e.g. for a
+//     property `foo`, there should be a definition
+//     `inline constexpr foo::value_t foo_v`.
+//  5. Specialize `sycl::is_property` and `sycl::is_property_of` for the
+//     property class.
```

Comment:
Since there are a lot of steps, maybe give some code example in the comments, or refer the reader to one that is properly implemented elsewhere.

Reply:
I have moved this explanation to the new file `sycl/include/sycl/ext/oneapi/property_list/properties.hpp`, added a how-to for adding runtime properties, and examples to both. What do you think?